pkg/csi/cinder/controllerserver.go

```diff
@@ -578,6 +578,14 @@ func (cs *controllerServer) ControllerExpandVolume(ctx context.Context, req *csi
 		return nil, status.Errorf(codes.Internal, fmt.Sprintf("Could not resize volume %q to size %v: %v", volumeID, volSizeGB, err))
 	}
 
+	// we need wait for the volume to be available or InUse, it might be error_extending in some scenario
+	targetStatus := []string{openstack.VolumeAvailableStatus, openstack.VolumeInUseStatus}
+	err = cs.Cloud.WaitVolumeTargetStatus(volumeID, targetStatus)
+	if err != nil {
+		klog.Errorf("Failed to WaitVolumeTargetStatus: %v", err)
+		return nil, status.Error(codes.Internal, fmt.Sprintf("[ControllerExpandVolume] failed to expand volume: %v", err))
```

Comment:
would be helpful , if volume_id is also added here.
:nit: status.Errorf could be used.

May be something like this in Error message would be goof.
 "[ControllerExpandVolume] Volume %s not in target state after resize operation : %v", volumeID, err))"

Reply:
make sense to me, updated